src/Chainweb/BlockHeader.hs

```diff
@@ -216,6 +217,23 @@ slowEpochGuard (ParentHeader p)
     | otherwise = False
 {-# INLINE slowEpochGuard #-}
 
+-- | Skip validation of feature flags for block heights up to 340000.
+--
+-- Unused feature flag bits are supposed to be set to 0. This was not enforced
+-- in chainweb-node versions <= 1.5. There is a large number of blocks in the
+-- history of mainnet before 2020-02-20, that have non-zero feature flags. In
+-- order to prepare future use of fleature flag feature flag validation will
+-- start at block height 340000.
+--
+-- This guard grandfathers the this behavior up to block height 340000.
+--
+-- Blockheight 340000 is expected to occur on mainnet01 on 2019-02-24.
+--
+skipFeatureFlagValidationGuard :: BlockHeader -> Bool
```

Comment:
This should be in `ChainwebVersion` with the goal of keeping our time-based switches in one place.

Reply:
Agreed, this guard section along with the documentation, originally pre-dates the dates in `Chainweb.Version`, we just forgot to move it when we started to put guards in `Chainweb.Version`.

I'll move it.